core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1054,15 +1053,17 @@ class DAGScheduler(
         return
     }
 
+    // Mark the stage as submitted and send an event on the bus. If there are pending tasks, submit
+    // them, otherwise immediately mark the stage as finished, and submit any child stages.
+    stage.latestInfo.submissionTime = Some(clock.getTimeMillis())
```

Comment:
why not just put this before the old listenerBus.post()  on L991?

this change also changes the behavior if there is a an exception while creating the tasks -- you no longer post a SparkListenerStageSubmitted.

I don't have any particular reason why you'd want the behavior one way or the other, but w/out an argument for actually changing the behavior, I'd rather do the minimal change.